modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -607,15 +607,22 @@ private void _updateDLSizeLimitConfiguration(
 
 		MimeTypeLimit[] mimeTypeLimits = settings.getMimeTypeLimits();
 
-		if (mimeTypeLimits != null) {
-			for (MimeTypeLimit mimeTypeLimit : settings.getMimeTypeLimits()) {
-				String mimeType = mimeTypeLimit.getMimeType();
-
-				if (Validator.isNotNull(mimeType)) {
-					mimeTypeSizeLimits.put(
-						mimeType,
-						GetterUtil.getLong(mimeTypeLimit.getMaximumSize()));
-				}
```

Comment:
In the original code you were modifying the `mimeTypeLimits` map, that comes directly from the configuration. Maybe this was the original issue? Even if the update fails, the map has already the wrong values in it.

What this code should have done is to create a copy of the original map, and modify the copy. Can you test that in the original version of the code?

Reply:
not needed, In the las PR i'm reseting the mimeTypeSizeLimits in line 617 in the needed cases